cpp/test/distance/pairwise_distance_c.cu

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda.h>
+
+#include <gtest/gtest.h>
+#include <raft/core/device_mdarray.hpp>
+#include <raft/core/handle.hpp>
+#include <raft/random/rng.cuh>
+
+#include <cuvs/distance/distance.h>
+
+extern "C" void run_pairwise_distance(int64_t n_rows,
+                                      int64_t n_queries,
+                                      int64_t n_dim,
+                                      float* index_data,
```

Comment:
What happens if someone were to compile these tests against a `double` pointer? It wouldn't cause the C++ to compile a new specialization would it? Ideally it would fail gracefull and state that it doesn't support the given type combinations.

Reply:
In cases where we are passing a tensor with an invalid dtype (float16 etc) - we should fail with an informative error message. (no new specializations will be compiled - will just use the existing float/double ones).